docs/modules/ROOT/pages/index.adoc

```diff
@@ -38,4 +64,308 @@ include::includes/quarkus-solace-extension-outgoing.adoc[leveloffset=+1, opts=op
+public class TemperaturesProducer {
+
+    private final Random random = new Random();
+
+    @Outgoing("temperatures-out")
+    public Multi<Double> generate() {
+        // Emit 1000 records
+        return Multi.createFrom().range(0, 1000)
+            .map(x -> random.nextDouble());
+    }
+
+}
+----
+
+You can also generate a `org.eclipse.microprofile.reactive.messaging.Message` with required metadata and publish to Solace.
+
+[source,java]
+----
+@ApplicationScoped
+public class TemperaturesProducer {
+    private final Random random = new Random();
+
+    @Outgoing("temperatures-out")
+    Multi<Message<Double>> publishTemperatures() {
+        return Multi.createFrom().range(0, 1000)
+            .map(i -> {
+                 SolaceOutboundMetadata outboundMetadata = SolaceOutboundMetadata.builder()
+                .setApplicationMessageId(Integer.toString(i)).createPubSubOutboundMetadata();
+                 return Message.of(random.nextDouble(), Metadata.of(outboundMetadata));
+            });
+    }
+}
+----
+
+*SolaceOutboundMetadata* allows to configure metadata for the message. It supports all the headers supported by Solace and custom user properties. In addition to this it also supports configuring dynamic topic which overrides the default topic in application configuration file.
+
+Generating `org.eclipse.microprofile.reactive.messaging.Message` with dynamic topic and publish to Solace.
+
+[source,java]
+----
+@ApplicationScoped
+public class TemperaturesProducer {
+    private final Random random = new Random();
+
+    @Outgoing("temperatures-out")
+    Multi<Message<Double>> publishTemperatures() {
+        return Multi.createFrom().range(0, 1000)
+            .map(i -> {
+                 SolaceOutboundMetadata outboundMetadata = SolaceOutboundMetadata.builder()
+                .setApplicationMessageId(Integer.toString(i))
+                .setDynamicDestination("device/" + Integer.toString(i) + "/temperature").createPubSubOutboundMetadata();
+                 return Message.of(random.nextDouble(), Metadata.of(outboundMetadata));
+            });
+    }
+}
+----
+
+Sending messages with __@Emitter__
+
+[source,java]
+----
+@Path("/temperatures")
+public class PublisherResource {
+
+    @Channel("temperatures-out")
+    MutinyEmitter<Double> temperatureEmitter;
+
+    @POST
+    @Path("/publish")
+    public Uni<Void> publish(Double temperature) {
+        return temperatureEmitter.send(person);
+    }
+}
+----
+
+== Producer Back-Pressure strategies
+
+Quarkus Solace Messaging connector provides three different strategies to handle back-pressure when publishing messages
+
+{empty}1.Reject - Publisher will start rejecting messages once specified limit is reached
+
+{empty}2.Wait - Publisher is throttled when a specified limit is reached
+
+{empty}3.Elastic - Use an unlimited internal buffer (default)
+
+CAUTION: In the current version we don't recommend to use back-pressure strategy `Reject` as it is in evolving phase.
```

Comment:
@ozangunalp I have added this caution here as we have open issue #29 on handling this scenario